datamapper/changeset.py

```diff
@@ -20,38 +21,81 @@
 from datamapper.model import Association, BelongsTo, Model
 
 FieldValidator = Callable[[Any], Union[bool, str]]
-Data = Union[dict, Model]
+Data = Union[Tuple[dict, dict], Model]
 
-T = TypeVar("T", dict, Model)
+T = TypeVar("T", Tuple[dict, dict], Model)
 
 
 class ChangesetDataWrapper(Generic[T]):
+    @staticmethod
+    def instance(data: T) -> ChangesetDataWrapper:
+        if isinstance(data, Model):
+            return ModelChangesetDataWrapper(data)
+        elif isinstance(data, tuple):
+            return DictChangesetDataWrapper(data)
+        else:
+            raise AttributeError("Changeset data must be a Model or (dict, dict).")
+
     def __init__(self, data: T):
-        self.data: T = data
+        self.data: T = data  # pragma: no cover
+
+    def get(self, field: str, default: Any = None) -> Optional[Any]:
+        return self.attributes.get(field, default)
 
     def apply_changes(self, changes: dict) -> T:
-        if isinstance(self.data, dict):
-            return dict_merge(self.data, changes)
-        elif isinstance(self.data, Model):
-            attrs = {**self.data.attributes, **changes}
-            return type(self.data)(**attrs)
+        raise NotImplementedError()  # pragma: no cover
 
     def field_type(self, field: str) -> Type:
-        if isinstance(self.data, Model):
-            column = self.data.__table__.columns.get(field)
-            return column.type if column is not None else None
-        else:
-            return type(self.data.get(field))
+        raise NotImplementedError()  # pragma: no cover
 
     def association(self, field: str) -> Association:
-        if not isinstance(self.data, Model):
-            raise ValueError("Data of type dict has no associations")
+        raise NotImplementedError()  # pragma: no cover
+
+    @property
+    def attributes(self) -> dict:
+        raise NotImplementedError()  # pragma: no cover
+
+    def __repr__(self) -> str:
+        return self.data.__repr__()  # pragma: no cover
+
+
+class DictChangesetDataWrapper(ChangesetDataWrapper):
```

Comment:
I'm still pretty confused by generics in Python, but should this be `ChangesetDataWrapper[dict]`?